### PR TITLE
Adding the tbasic message and trace logs parameter

### PR DESCRIPTION
**Pull request by tonyreigns**
Providing format option for messages.log that looks like WAS basic. The new format option is called tbasic. 
tbasic is also added as an alias for the basic trace option.

#17413

### CODE REVIEW

**Comment by tonyreigns** [comment]
#build

**Comment by fmhwong** [comment]
Please update the 2nd copyright year to 2021.

**Review by ManasiGandhi — APPROVED** [review]
Looks good to me.

**Comment by skasund** [comment]
L2 message review not required

**Comment by tonyreigns** [comment]
#build

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java [hunk 1]
```diff
@@ -908,6 +1003,26 @@ private void fixedClassString(StringBuilder output, String s, int len) {
...
+        int i = s.lastIndexOf('.');
+        if (i >= 0) {
+            s = s.substring(i + 1);
+        }
+
+        if (s.length() > len) {
+            output = s.substring(0, len);
+        } else {
+            output = s;
+            if (len > s.length())
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java]: Not sure why we need to compare the length again.

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/ConsoleLogHandler.java [hunk 2]
```diff
@@ -236,7 +247,7 @@ public void setConsoleLogLevel(Integer consoleLogLevel) {
     }
 
     /**
-     * The format to set (i.e. DEV, SIMPLE, or JSON)
+     * The format to set (i.e. DEV, SIMPLE, TBASIC or JSON)
      *
      * @param format the format to set (i.e. DEV, SIMPLE, or JSON)
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/ConsoleLogHandler.java]: Missing TBASIC in the i.e. bracket?

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/MessageLogHandler.java [hunk 3]
```diff
@@ -97,7 +100,7 @@ public void setBasicFormatter(BaseTraceFormatter formatter) {
     }
 
     /**
-     * The format to set (i.e. SIMPLE, JSON, or BASIC (deprecated))
+     * The format to set (i.e. SIMPLE, JSON, TBASIC or BASIC (deprecated))
      *
      * @param format the format to set (i.e. SIMPLE, JSON, or BASIC (deprecated))
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/MessageLogHandler.java]: Missing TBASIC?

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java [hunk 4]
```diff
@@ -610,6 +665,26 @@ private String createFormattedString(LogRecord logRecord, Object id, String txt)
                     sb.append(method);
                 sb.append(' '); // yes, this space is always there.
 
+                // append formatted message -- includes formatted args
+                sb.append(txt);
+                if (stackTrace != null)
+                    sb.append(nlBasicPadding).append(stackTrace);
+                break;
+            case TBASIC:
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java:685]: Is this case block exactly same as BASIC? 

If yes, you can use

```
case BASIC:
case TBASIC:
```

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java [hunk 5]
```diff
@@ -782,6 +857,26 @@ public String traceFormatGenData(GenericData genData) {
...
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java:858]: Is this case block exactly same as BASIC? 

If yes, you can use

```
case BASIC:
case TBASIC:
```

**Review by fmhwong — CHANGES_REQUESTED** [review]
Please add tests for the new format.

**Comment by tonyreigns** [comment]
#build

**Comment by tonyreigns** [comment]
#build

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/ConsoleFormatTest.java [hunk 6]
```diff
@@ -59,13 +59,15 @@
...
     private static final String DEPRECATED_BASIC_FORMAT = "basic";
+    private static final String TBASIC_FORMAT = "tbasic";
     private static final String INVALID_CONSOLE_FORMAT = "simples";
 
     private static final String SERVER_NAME = "com.ibm.ws.logging.consoleformat";
     private static final String SERVER_ENV_NAME = "com.ibm.ws.logging.consoleformatenv";
 
     private static final String DEV_FORMAT_REGEX_PATTERN = "([A-Z]{3,}   )"; // Matches with line that has [<LOG_LEVEL>   ] in the beginning. e.g. [AUDIT   ]
     private static final String SIMPLE_FORMAT_REGEX_PATTERN = "(\\[\\d{1,4}.*)"; // Matches with line that [23/02/20 ... ] in the beginning.
+    private static final String TBASIC_FORMAT_REGEX_PATTERN = "([a-zA-Z0-9- ]{8} [aA-zZ]{1,13}\\s{48})";
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/ConsoleFormatTest.java]: This is still checking for 48 spaces but we have removed that long spaces.

**Comment by tonyreigns** [comment]
#build

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/ConsoleFormatTest.java [hunk 7]
```diff
@@ -59,13 +59,16 @@
...
     private static final String DEPRECATED_BASIC_FORMAT = "basic";
+    private static final String TBASIC_FORMAT = "tbasic";
     private static final String INVALID_CONSOLE_FORMAT = "simples";
 
     private static final String SERVER_NAME = "com.ibm.ws.logging.consoleformat";
     private static final String SERVER_ENV_NAME = "com.ibm.ws.logging.consoleformatenv";
 
     private static final String DEV_FORMAT_REGEX_PATTERN = "([A-Z]{3,}   )"; // Matches with line that has [<LOG_LEVEL>   ] in the beginning. e.g. [AUDIT   ]
     private static final String SIMPLE_FORMAT_REGEX_PATTERN = "(\\[\\d{1,4}.*)"; // Matches with line that [23/02/20 ... ] in the beginning.
+    private static final String TBASIC_FORMAT_REGEX_PATTERN = "([a-zA-Z0-9- ]{8} [aA-zZ]{1,13} [aA-zZ]{1}\\s{3})";
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/ConsoleFormatTest.java]: This regex failed to match this valid line. 
```
[17/03/21 13:42:33:333 EDT] 00000063 Angry         W   BADAP0001W: quit bugging me.
```

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/ConsoleFormatTest.java [hunk 8]
```diff
@@ -59,13 +59,16 @@
...
+    private static final String TBASIC_FORMAT_REGEX_PATTERN_CLASS = "([a-zA-Z0-9- ]{8} [aA-zZ]{1,13} [aA-zZ]{1}\\s{1})";
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/ConsoleFormatTest.java]: This regex failed to match this valid line.
```
[17/03/21 13:42:33:328 EDT] 00000063 servlet       I com.ibm.ws.webcontainer.servlet.ServletWrapper init SRVE0242I: [badapp] [/badapp] [com.ibm.ws.lumberjack.badness.Angry]: Initialization successful.
```

File: dev/com.ibm.ws.logging_fat/publish/servers/com.ibm.ws.logging.tbasicheaderformat/bootstrap.properties [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+###############################################################################
+bootstrap.include=../testports.properties
+com.ibm.ws.logging.hideMessage=CWWKZ0058I
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging_fat/publish/servers/com.ibm.ws.logging.tbasicheaderformat/bootstrap.properties]: Do we need this property `com.ibm.ws.logging.hideMessage=CWWKZ0058I`?

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/ConsoleFormatTest.java (hunk 7 above)
Comment (tonyreigns) [reply dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/ConsoleFormatTest.java]: Resolved.

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/ConsoleFormatTest.java (hunk 8 above)
Comment (tonyreigns) [reply dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/ConsoleFormatTest.java]: Resolved.

File: dev/com.ibm.ws.logging_fat/publish/servers/com.ibm.ws.logging.tbasicheaderformat/bootstrap.properties (hunk 9 above)
Comment (tonyreigns) [reply dev/com.ibm.ws.logging_fat/publish/servers/com.ibm.ws.logging.tbasicheaderformat/bootstrap.properties]: No, resolved.

**Comment by tonyreigns** [comment]
#build

**Comment by tonyreigns** [comment]
#build

File: dev/com.ibm.ws.logging.osgi/resources/OSGI-INF/metatype/metatype.xml [hunk 10]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
-    Copyright (c) 2011, 2020 IBM Corporation and others.
+    Copyright (c) 2011, 2021 IBM Corporation and others.
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging.osgi/resources/OSGI-INF/metatype/metatype.xml:3]: No need to change the copyright year as there is no change in the file.

File: dev/com.ibm.ws.logging/resources/OSGI-INF/metatype/metatype.xml [hunk 11]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
-    Copyright (c) 2011, 2020 IBM Corporation and others.
+    Copyright (c) 2011, 2021 IBM Corporation and others.
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging/resources/OSGI-INF/metatype/metatype.xml:3]: No need to change the copyright year as there is no change in the file.

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java [hunk 12]
```diff
@@ -548,6 +554,86 @@ public String messageLogFormat(GenericData genData) {
         return sb.toString();
     }
 
+    public Boolean betaFenceCheck() throws UnsupportedOperationException {
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java]: This method doesn't throw UnsupportedOperationException.

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java [hunk 13]
```diff
@@ -71,6 +75,8 @@ public Integer run() {
     static final String SYSOUT = "SystemOut";
     static final String SYSERR = "SystemErr";
 
+    public static Boolean isBetaEdition;
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java]: Change to private

File: dev/com.ibm.ws.logging/resources/OSGI-INF/metatype/metatype.xml (hunk 11 above)
Comment (tonyreigns) [reply dev/com.ibm.ws.logging/resources/OSGI-INF/metatype/metatype.xml:3]: Resolved. It was leftover after the metatype documentation was temporarily removed.

**Comment by tonyreigns** [comment]
#build

**Review by fmhwong — APPROVED** [review]
LGTM

File: dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/ConsoleFormatTest.java [hunk 14]
```diff
@@ -307,6 +331,29 @@ public void testSimpleConsoleFormatWithSysOutSysErrMsgs() throws Exception {
...
+
+        // Set the consoleFormat="tbasic", traceSpec=off, isoDateFormat=false in server.xml
+        setServerConfiguration(server, TBASIC_FORMAT, false, false, consoleLogFile);
+
+
+        // Run application to generate Audit messages including the method and class name using logger.logp
+        hitWebPage("tbasic-servlet", "TBasicServlet", false, null);
+
+        // Verify the Audit message
+        String line = server.waitForStringInLog(" helloMethod ", consoleLogFile);
```
Comment (fmhwong) [inline dev/com.ibm.ws.logging_fat/fat/src/com/ibm/ws/logging/fat/ConsoleFormatTest.java:349]: Please add a null check.

**Review by fmhwong — CHANGES_REQUESTED** [review]
Please fix the test failures.

**Comment by tonyreigns** [comment]
#build

**Comment by tonyreigns** [comment]
#build

**Comment by tonyreigns** [comment]
#build

**Review by fmhwong — APPROVED** [review]
LGTM

**Comment by tonyreigns** [comment]
#run-libby-bot

**Pull request merged**